File: extraction/Haskell/enum.hs
module Enum where

import qualified Prelude

data Bool =
   True
 | False

negb :: Bool -> Bool
negb b =
  case b of {
   True -> False;
   False -> True}

andb :: Bool -> Bool -> Bool
andb b1 b2 =
  case b1 of {
   True -> b2;
   False -> False}

orb :: Bool -> Bool -> Bool
orb b1 b2 =
  case b1 of {
   True -> True;
   False -> b2}

nandb :: Bool -> Bool -> Bool
nandb b1 b2 =
  case andb b1 b2 of {
   True -> False;
   False -> True}

andb3 :: Bool -> Bool -> Bool -> Bool
andb3 b1 b2 b3 =
  andb (andb b1 b2) (andb b2 b3)

File: extraction/Ocaml/enum.ml

type bool =
| True
| False

(** val negb : bool -> bool **)

let negb = function
| True -> False
| False -> True

(** val andb : bool -> bool -> bool **)

let andb b1 b2 =
  match b1 with
  | True -> b2
  | False -> False

(** val orb : bool -> bool -> bool **)

let orb b1 b2 =
  match b1 with
  | True -> True
  | False -> b2

(** val nandb : bool -> bool -> bool **)

let nandb b1 b2 =
  match andb b1 b2 with
  | True -> False
  | False -> True

(** val andb3 : bool -> bool -> bool -> bool **)

let andb3 b1 b2 b3 =
  andb (andb b1 b2) (andb b2 b3)
